From 09a3e0dd226ce41deeb78b3a25189c1012483e99 Mon Sep 17 00:00:00 2001
From: smethnani <82812166+smethnani@users.noreply.github.com>
Date: Sun, 30 Nov 2025 21:34:15 +0000
Subject: [PATCH] Get trained data from jfrog

---
 package.json           |  3 ++-
 scripts/postinstall.js | 56 ++++++++++++++++++++++++++++++++++++------
 2 files changed, 50 insertions(+), 9 deletions(-)

diff --git a/package.json b/package.json
index d58173d..b982d7e 100644
--- a/package.json
+++ b/package.json
@@ -18,6 +18,7 @@
     "postinstall": "node ./scripts/postinstall.js"
   },
   "dependencies": {
+    "axios": "^0.30.0",
     "babel-polyfill": "^6.26.0",
     "cron-validator": "^1.1.1",
     "elastic-builder": "^2.7.1",
@@ -97,4 +98,4 @@
     "sha.js": "^2.4.12",
     "pbkdf2": "^3.1.3"
   }
-}
\ No newline at end of file
+}
diff --git a/scripts/postinstall.js b/scripts/postinstall.js
index 1e53f1c..a7aa36a 100644
--- a/scripts/postinstall.js
+++ b/scripts/postinstall.js
@@ -5,9 +5,10 @@
 
 // @ts-check
 // workaround for Safari support before https://github.com/niklasvh/html2canvas/pull/2911 is merged
-const https = require('https');
+const axios = require('axios');
 const fs = require('fs');
 const replace = require('replace-in-file');
+const { HttpsProxyAgent } = require('https-proxy-agent');
 
 const options = {
   files: [
@@ -35,14 +36,53 @@ try {
 }
 
 // download tesseract model
-const modelFile = fs.createWriteStream(__dirname + '/../common/tesseract/eng.traineddata.gz');
-https.get(
-  'https://raw.githubusercontent.com/naptha/tessdata/gh-pages/4.0.0_best/eng.traineddata.gz',
-  function (response) {
-    response.pipe(modelFile);
+const proxyOptions = {};
+if (process.env.PROXYHOST) {
+  const proxyUrl = process.env.PROXYHOST.startsWith('http')
+    ? process.env.PROXYHOST
+    : `http://${process.env.PROXYHOST}`;
+
+  console.log(`HTTPS proxy is set to: ${proxyUrl}`);
+  const proxyAgent = new HttpsProxyAgent(proxyUrl);
+  proxyOptions.httpsAgent = proxyAgent;
+}
+
+if (!process.env.ARTIFACTORY_USER || !process.env.ARTIFACTORY_PASSWORD) {
+  console.error('Missing Artifactory credentials.');
+  process.exit(1);
+}
+
+const ARTIFACTORY_USER = process.env.ARTIFACTORY_USER;
+const ARTIFACTORY_PASSWORD = process.env.ARTIFACTORY_PASSWORD;
+
+const auth = `${ARTIFACTORY_USER}:${ARTIFACTORY_PASSWORD}`;
+const encodedAuth = Buffer.from(auth).toString('base64');
+
+const url =
+  'https://canonical.jfrog.io/artifactory/dataplatform-generic-stable-local/tessdata/eng.traineddata.gz';
+const outputFile = __dirname + '/../common/tesseract/eng.traineddata.gz';
+
+axios
+  .get(url, {
+    responseType: 'stream',
+    headers: {
+      Authorization: `Basic ${encodedAuth}`,
+    },
+    ...proxyOptions,
+  })
+  .then((response) => {
+    const modelFile = fs.createWriteStream(outputFile);
+    response.data.pipe(modelFile);
+
     modelFile.on('finish', () => {
       modelFile.close();
       console.log('Downloaded eng.traineddata.gz for tesseract.js');
     });
-  }
-);
+
+    modelFile.on('error', (err) => {
+      console.error('Error writing file:', err);
+    });
+  })
+  .catch((err) => {
+    console.error('Download failed:', err.message);
+  });
-- 
2.49.0

