diff --git a/scripts/postinstall.js b/scripts/postinstall.js
index 1e53f1c..a7aa36a 100644
--- a/scripts/postinstall.js
+++ b/scripts/postinstall.js
@@ -5,9 +5,10 @@
 
 // @ts-check
 // workaround for Safari support before https://github.com/niklasvh/html2canvas/pull/2911 is merged
-const https = require('https');
+const axios = require('axios');
 const fs = require('fs');
 const replace = require('replace-in-file');
+const { HttpsProxyAgent } = require('https-proxy-agent');
 
 const options = {
   files: [
@@ -35,14 +36,55 @@ try {
 }
 
 // download tesseract model
-const modelFile = fs.createWriteStream(__dirname + '/../common/tesseract/eng.traineddata.gz');
-https.get(
-  'https://raw.githubusercontent.com/naptha/tessdata/gh-pages/4.0.0_best/eng.traineddata.gz',
-  function (response) {
-    response.pipe(modelFile);
+const proxyOptions = {};
+const proxyEnv = process.env.https_proxy || process.env.http_proxy;
+if (proxyEnv) {
+  const proxyUrl = proxyEnv.startsWith('http')
+    ? proxyEnv
+    : `http://${proxyEnv}`;
+
+  console.log(`Proxy is set to: ${proxyUrl}`);
+  const proxyAgent = new HttpsProxyAgent(proxyUrl);
+  proxyOptions.httpsAgent = proxyAgent;
+}
+
+if (!process.env.ARTIFACTORY_USER || !process.env.ARTIFACTORY_PASSWORD) {
+  console.error('Missing Artifactory credentials.');
+  process.exit(1);
+}
+
+const ARTIFACTORY_USER = process.env.ARTIFACTORY_USER;
+const ARTIFACTORY_PASSWORD = process.env.ARTIFACTORY_PASSWORD;
+
+const auth = `${ARTIFACTORY_USER}:${ARTIFACTORY_PASSWORD}`;
+const encodedAuth = Buffer.from(auth).toString('base64');
+
+const url = process.env.ARTIFACTORY_URL
+  ? `${process.env.ARTIFACTORY_URL}/tessdata/eng.traineddata.gz`
+  : 'https://raw.githubusercontent.com/naptha/tessdata/gh-pages/4.0.0_best/eng.traineddata.gz';
+const outputFile = __dirname + '/../common/tesseract/eng.traineddata.gz';
+
+axios
+  .get(url, {
+    responseType: 'stream',
+    headers: {
+      Authorization: `Basic ${encodedAuth}`,
+    },
+    ...proxyOptions,
+  })
+  .then((response) => {
+    const modelFile = fs.createWriteStream(outputFile);
+    response.data.pipe(modelFile);
+
     modelFile.on('finish', () => {
       modelFile.close();
       console.log('Downloaded eng.traineddata.gz for tesseract.js');
     });
-  }
-);
+
+    modelFile.on('error', (err) => {
+      console.error('Error writing file:', err);
+    });
+  })
+  .catch((err) => {
+    console.error('Download failed:', err.message);
+  });
