diff --git a/run.sh b/run.sh
index 640672e6..3561058e 100755
--- a/run.sh
+++ b/run.sh
@@ -11,14 +11,8 @@ set -e
 
 if [ -z "$1" ]; then (echo "syntax: run.sh [workflow.py]"; exit -1); fi
 command -v python3 >/dev/null 2>&1 || (echo "missing python3"; exit -1)
-command -v pip >/dev/null 2>&1 || (echo "missing python3-pip"; exit -1)
-command -v pipenv >/dev/null 2>&1 || (echo "missing pipenv"; exit -1)
 
 DIR="$(dirname "$0")"
 
-echo "Installing dependencies in $DIR ..."
-export PIPENV_PIPFILE="$DIR/Pipfile"
-python3 -m pipenv install
-
-if [[ ! "$1" == *benchmark* ]]; then (echo "Running "$1" ${@:2} ..."; ); fi
-python3 -m pipenv run python "$1" ${@:2}
+echo "Running "$1" ${@:2} ..."
+python3 "$1" ${@:2}
